README.md

```diff
@@ -1,14 +1,133 @@
+
+In the Charmed Apache Kafka Connect operator, adding a plugin is as simple as calling the `juju attach-resource` command. Make sure that you bundle all required JAR files into a single TAR archive (let's call that `my-plugin.tar`) and then use the following command:
+
+```shell
+juju attach-resource kafka-connect connect-plugin=./my-plugin.tar
+```
+
+This would trigger a restart of `kafka-connect` charm, once all units show as `active|idle`, your desired plugin is ready to use. 
+
+There is no limit on the number of plugins that could be manually added. However, the recommended way for common use-cases of ETL tasks on Data Platform charmed operators is by using the [Template Connect Integrator](https://github.com/canonical/template-connect-integrator) charm.
+
+## Relations
+
+The Charmed Apache Kafka Connect Operator supports Juju [relations](https://juju.is/docs/olm/relations) for interfaces listed below.
+
+#### `connect_client` interface:
+
+The `connect_client` interface is used with any requirer/integrator charm adhering to the `connect-client` charm relation interface. Integrators will automatically handle connectors/tasks lifecycle on Apache Kafka Connect including plugin management, startup, cleanup, and scaling, and simplify common ETL operations on Data Platform line of products.
+
+A curated set of integrators for common ETL use cases on Canonical Data Platform line of products are provided in the [Template Connect Integrator](https://github.com/canonical/template-connect-integrator) repository. These cover use-cases such as loading data from/to MySQL, PostgreSQL, Opensearch, S3-compliant storage services, and active/passive replication of Apache Kafka topics using MirrorMaker.
+
+#### `tls-certificates` interface:
 
-Charmed Apache Kafka Connect is free software, distributed under the Apache Software License, version 2.0. See [LICENSE](https://github.com/canonical/kafka-operator/blob/main/LICENSE) for more information.
+The `tls-certificates` interface could be used with any charm that adheres to [`tls-certifcates`](https://github.com/canonical/charm-relation-interfaces/tree/main/docs/json_schemas/tls_certificates/v1) **provider** charm relation interface. One example is the [`self-signed-certificates`](https://github.com/canonical/self-signed-certificates-operator) operator by Canonical.
 
+Note that TLS could be enabled in three different modes:
+- Only on Apache Kafka Connect REST interface.
```

Comment:
Nitpick: Add an empty line before a list.
Suggestion: Improved wording for list elements.

```suggestion

- For Apache Kafka Connect REST interface only
```